be1-go/hub/organizer.go

```diff
@@ -594,7 +590,7 @@ func (c *laoChannel) processMessageObject(public message.PublicKey, data message
 			c.inboxMu.Unlock()
 			return nil
 		}
-		msg.WitnessSignatures = append(msg.WitnessSignatures, message.PublicKeySignaturePair{
+		msgInfo.message.WitnessSignatures = append(msgInfo.message.WitnessSignatures, message.PublicKeySignaturePair{
```

Comment:
I think this wouldn't work ebcause `msgInfo` is not a pointer to the struct.

Please have a look at: https://play.golang.org/p/2uHpMO2lCzh

Please try declaring `inbox` as `map[string]*messageInfo` instead

Reply:
At this stage it might be interesting to create a new inbox type:

```
type inbox {
  sync.Mutex
  msgs map[string]*messageInfo
}

func (i inbox) addWitnessSig(key string, sig ...) {
  i.msgs[key] = append(i.msgs[key], sig)
}
```

then

```
sig := message.PublicKeySignaturePair{
  ...
}
c.inbox.addWitnessSig(msgEncoded, sig)